Vuzix/app/src/main/java/com/facilitation/view/MainActivity.kt

```diff
@@ -1,46 +1,106 @@
 package com.facilitation.view
 
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
 import android.os.Bundle
-import androidx.activity.ComponentActivity
-import androidx.activity.compose.setContent
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.material3.MaterialTheme
-import androidx.compose.material3.Surface
-import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
-import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
-import com.facilitation.view.ui.theme.ViewTheme
-
-class MainActivity : ComponentActivity() {
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.TextView
+import android.widget.Toast
+import com.vuzix.hud.actionmenu.ActionMenuActivity
+
+
+class MainActivity : ActionMenuActivity() {
+    var HelloMenuItem: MenuItem? = null
+    var SpotifyMenuItem: MenuItem? = null
+    var mainText: TextView? = null
+    private var broadcastReceiver: BroadcastReceiver? = null
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContent {
-            ViewTheme {
-                // A surface container using the 'background' color from the theme
-                Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colorScheme.background
-                ) {
-                    Greeting("Android")
+        setContentView(R.layout.activity_main)
+        initializeBroadcastReceiver()
+    }
+
+    override fun onCreateActionMenu(menu: Menu): Boolean {
+        super.onCreateActionMenu(menu)
+        menuInflater.inflate(R.menu.menu, menu)
+        HelloMenuItem = menu.findItem(R.id.item1)
+        SpotifyMenuItem = menu.findItem(R.id.item2)
+        mainText = findViewById(R.id.mainTextView)
+        updateMenuItems()
+        return true
+    }
+    override fun alwaysShowActionMenu(): Boolean {
+        return false
+    }
+
+    private fun updateMenuItems() {
+        if (HelloMenuItem == null) {
+            return
+        }
+        SpotifyMenuItem?.isEnabled = false
+    }
+
+    //Action Menu Click events
+    //This events where register via the XML for the menu definitions.
+    fun showHello(item: MenuItem?) {
+        val broadcastIntent = Intent("com.facilitation.view.GET")
+        broadcastIntent.putExtra("broadcastMessage", "Goodbye World!")
+        sendBroadcast(broadcastIntent)
```

Comment:
Is this being sent when you get the GET action? If so, should I receive it?

Reply:
this was for testing internal broadcasts before you broadcasted from the phone